### PR TITLE
Add auto instrumentation to twig, and opensearch and kernel

### PR DESCRIPTION
**Pull request by Gaitholabi**

### CODE REVIEW

File: src/Instrumentation/OpenSearchInstrumentation.php [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+use OpenTelemetry\API\Trace\Span;
+use OpenTelemetry\API\Trace\SpanKind;
+use OpenTelemetry\API\Trace\StatusCode;
+use OpenTelemetry\Context\Context;
+use OpenTelemetry\SemConv\TraceAttributes;
+use Throwable;
+
+use function OpenTelemetry\Instrumentation\hook;
+
+class OpenSearchInstrumentation
```
Comment (shyim) [inline src/Instrumentation/OpenSearchInstrumentation.php:18]: can we make opensearch optional? register this instrumentation only when opensearch client class exists and move it to require-dev so phpstan is still happy?

File: _register.php [hunk 2]
```diff
@@ -6,8 +6,11 @@
...
 use Shopware\OpenTelemetry\Instrumentation\DALInstrumentation;
 use Shopware\OpenTelemetry\Instrumentation\HttpClientInstrumentation;
+use Shopware\OpenTelemetry\Instrumentation\OpenSearchInstrumentation;
+use Shopware\OpenTelemetry\Instrumentation\ShopwareInstrumentation;
 use Shopware\OpenTelemetry\Instrumentation\SymfonyInstrumentation;
 use OpenTelemetry\SDK\Sdk;
+use Shopware\OpenTelemetry\Instrumentation\TwigInstrumentation;
 
 if (class_exists(Sdk::class) && Sdk::isInstrumentationDisabled(SymfonyInstrumentation::NAME) === true) {
     return;
```
Comment (Gaitholabi) [inline _register.php:16]: I'm not convinced that `SymfonyInstrumentation::NAME` is correct, and it should be 'shopware'.

but for now, I think it's fine to opt for leaving it as is.. not to break BC

Comment (shyim) [reply _register.php:16]: you can change it. :)

File: _register.php [hunk 3]
```diff
@@ -6,10 +6,13 @@
...
 use Shopware\OpenTelemetry\Instrumentation\DALInstrumentation;
 use Shopware\OpenTelemetry\Instrumentation\HttpClientInstrumentation;
+use Shopware\OpenTelemetry\Instrumentation\OpenSearchInstrumentation;
+use Shopware\OpenTelemetry\Instrumentation\ShopwareInstrumentation;
 use Shopware\OpenTelemetry\Instrumentation\SymfonyInstrumentation;
 use OpenTelemetry\SDK\Sdk;
+use Shopware\OpenTelemetry\Instrumentation\TwigInstrumentation;
 
-if (class_exists(Sdk::class) && Sdk::isInstrumentationDisabled(SymfonyInstrumentation::NAME) === true) {
+if (class_exists(Sdk::class) && Sdk::isInstrumentationDisabled(ShopwareInstrumentation::NAME) === true) {
```
Comment (h1k3r) [inline _register.php:15]: This can enable disabled instrumentations. Need to communicate this change during release.

File: composer.json [hunk 4]
```diff
@@ -12,19 +12,21 @@
...
         "ext-opentelemetry": "*",
         "monolog/monolog": "2.* || 3.*",
         "shopware/core": "*",
-        "open-telemetry/api": "^1.0.0beta10",
+        "open-telemetry/api": "^1.0.3",
         "open-telemetry/sem-conv": "^1.22",
         "symfony/http-kernel": "*",
-        "symfony/http-client-contracts": "*"
+        "symfony/http-client-contracts": "*",
+        "open-telemetry/sdk": "^1.0"
```
Comment (h1k3r) [inline composer.json:25]: Do we really need an sdk? It seems it can be an optional dependency.

**Pull request closed without merging**